src/Illuminate/Validation/Validator.php

```diff
@@ -969,6 +969,25 @@ protected function validateInArray($attribute, $value, $parameters)
     }
 
     /**
+     * Validate that the current logged in user's password matches the given value.
+     *
+     * @param  string  $attribute
+     * @param  mixed   $value
+     * @return bool
+     */
+    protected function validatePassword($attribute, $value)
+    {
+        $auth = $this->container->make('auth');
+        $hasher = $this->container->make('hash');
+
+        if ($auth->guest()) {
+            return false;
+        }
+
+        return $hasher->check($value, $auth->user()->getAuthPassword());
```

Comment:
Doesn't this only use the default Auth guards, what if you need to validate on different auth guards. E.g `Auth::guard('foo')->user()->getAuthPassword()` etc.